app/src/main/AndroidManifest.xml

```diff
@@ -49,6 +49,12 @@
             android:enabled="true" />
 
         <receiver android:name=".Receivers.NotificationAlarmReceiver" />
+        <receiver android:name=".Receivers.NetworkDialogHandler">
```

Comment:
When you register it in the Manifest it gets activated everytime the network changes, even if your app isn't working or killed by the system, by registering in the manifest you are infact making the phone do more work even when the app is not active. Do you know why wasn't it working? Its a broadcast receiver only getting triggered when the network changes. So your testing steps should be to switch off network, startup the app and then switch the network back on again.